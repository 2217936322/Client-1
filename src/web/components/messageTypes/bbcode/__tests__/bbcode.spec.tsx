import React from 'react';
import BBCode from '@web/components/messageTypes/bbcode/__all__';
import { render } from '@testing-library/react';

describe('bbcode render', () => {
  test.each([
    [':cat:'],
    ['[img]http://example.com[/img]'],
    ['mix text image[img]http://example.com[/img]'],
    ['some sample text'],
    ['[asd]unrecognized tag should be ignore[/asd]'],
    ['http://baidu.com'],
    ['[url]http://baidu.com[/url]'],
    ['[url=http://baidu.com]百度[/url]'],
  ])('render "%s"', (originText) => {
    const { asFragment } = render(<BBCode plainText={originText} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
