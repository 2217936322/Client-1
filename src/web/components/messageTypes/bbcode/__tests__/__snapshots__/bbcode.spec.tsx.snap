// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bbcode render render ":cat:" 1`] = `
<DocumentFragment>
  <pre
    style="display: inline;"
  >
    🐱
  </pre>
</DocumentFragment>
`;

exports[`bbcode render render "[asd]unrecognized tag should be ignore[/asd]" 1`] = `
<DocumentFragment>
  <pre
    style="display: inline;"
  >
    unrecognized, ,tag, ,should, ,be, ,ignore
  </pre>
</DocumentFragment>
`;

exports[`bbcode render render "[img]http://example.com[/img]" 1`] = `
<DocumentFragment>
  <div
    class="image-viewer"
  >
    <mock-image
      role="chatimage"
      src="http://example.com"
    />
  </div>
</DocumentFragment>
`;

exports[`bbcode render render "[url=http://baidu.com]百度[/url]" 1`] = `
<DocumentFragment>
  <a
    href="http://baidu.com"
    rel="noopener"
    target="_blank"
    title="百度"
  >
    百度
  </a>
</DocumentFragment>
`;

exports[`bbcode render render "[url]http://baidu.com[/url]" 1`] = `
<DocumentFragment>
  <a
    href="http://baidu.com"
    rel="noopener"
    target="_blank"
    title="http://baidu.com"
  >
    http://baidu.com
  </a>
</DocumentFragment>
`;

exports[`bbcode render render "http://baidu.com" 1`] = `
<DocumentFragment>
  <a
    href="http://baidu.com"
    rel="noopener"
    target="_blank"
    title="http://baidu.com"
  >
    http://baidu.com
  </a>
</DocumentFragment>
`;

exports[`bbcode render render "mix text image[img]http://example.com[/img]" 1`] = `
<DocumentFragment>
  <pre
    style="display: inline;"
  >
    mix text image
  </pre>
  <div
    class="image-viewer"
  >
    <mock-image
      role="chatimage"
      src="http://example.com"
    />
  </div>
</DocumentFragment>
`;

exports[`bbcode render render "some sample text" 1`] = `
<DocumentFragment>
  <pre
    style="display: inline;"
  >
    some sample text
  </pre>
</DocumentFragment>
`;
